### ISSUE TITLE
Mask performance on multiple objects of same class in one bounding box

### ISSUE THREAD

**Issue by ridasalam** [dbolya/yolact#128]
<img width="345" alt="mask_issue" src="https://user-images.githubusercontent.com/49155016/63947953-278dc700-ca78-11e9-8bd6-02a8359ca336.png">

Hi, I have this issue when two objects of the same class appear in one bb (black one), the mask sometimes also includes some pixels from the other object (red). Do you have any recommendations for fixing this?

**Comment by dbolya** [comment]
Hmm yeah one of the limitations of our method is that kind of error is more prone to happen than say mask r-cnn. We can try to mitigate it though.

Have you tried fine tuning (#36)? That might give the masks more prior knowledge on how to avoid that. Other than that, you could also try messing with the hyper parameters like increasing the number of prototypes (k). To change the number of prototypes, in the config for protonet, change the last tuple's first number from 32 to something higher (it doesn't overfit or anything so you can try as high as you have memory for, though I wouldn't go over 256). We didn't observe this to change much in COCO, but it could change things in a smaller dataset.

**Comment by ridasalam** [comment]
Ok, ill look into changing the prototypes (already tried fine-tuning). How did you choose 32?

**Comment by dbolya** [comment]
As in the paper, it was the fastest that didn't decrease in performance. After k=16, the network started to decrease in performance (the updated table to include k=8 and k=16 will be coming soon).

**Comment by dbolya** [comment]
Since this has been open so long, I'm going to close it. Feel free to reopen if you have any updates.
